unit_testing/cli/test_cli.py

```diff
@@ -0,0 +1,27 @@
+from spinalcordtoolbox.utils import __sct_dir__
+import os
+import pytest
+
+scripts = []
+for line in open(os.path.join(__sct_dir__, 'scripts.txt')):
+    li = line.strip()
+    if not li.startswith("#"):
+        scripts.append(li)
+
+# Exclude scripts where no arguments is valid usage
+no_args_scripts = [s for s in scripts if s not in [
+    'isct_test_ants',          # No args -> tests ants binaries
+    'sct_testing',             # No args -> runs full test suite
+    'sct_check_dependencies',  # No args -> checks dependencies
+    'sct_version'              # No args -> prints version
+]]
+
+
+@pytest.mark.script_launch_mode('subprocess')
+@pytest.mark.parametrize("script", no_args_scripts)
+def test_argparse_no_arguments(script, script_runner):
+    """Ensure all scripts return error code and show usage descriptions
+    when called with no arguments."""
+    ret = script_runner.run(script)
+    assert ret.returncode is 2
+    assert 'usage' in ret.stdout.lower() or 'usage' in ret.stderr.lower()
```

Comment:
Hm I wonder if this would help: https://docs.pytest.org/en/stable/capture.html#accessing-captured-output-from-a-test-function

```suggestion
@pytest.mark.script_launch_mode('subprocess')
@pytest.mark.parametrize("script", no_args_scripts)
def test_argparse_no_arguments(capfd, script, script_runner):
    """Ensure all scripts return error code and show usage descriptions
    when called with no arguments."""
    ret = script_runner.run(script)
    assert ret.returncode is 2
    out, err = capfd.readouterr
    assert 'usage' in out.lower() or 'usage' in err.lower()
```

If so, then maybe you can avoid subprocess (which is probably slow) with

```suggestion
@pytest.mark.parametrize("script", no_args_scripts)
def test_argparse_no_arguments(capfd, script, script_runner):
    """Ensure all scripts return error code and show usage descriptions
    when called with no arguments."""
    ret = script_runner.run(script)
    assert ret.returncode is 2
    out, err = capfd.readouterr
    assert 'usage' in out.lower() or 'usage' in err.lower()
```

? maybe?

Reply:
I dunno, does that help at all?

Reply:
I implemented a modified version of this using our `main()` function calling in  f94200c. Test went from ~50s to ~5s. :smile:

Reply:
wow that's crazy! is it because it bypasses all the re-imports